configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionSerializer.java

```diff
@@ -54,6 +56,7 @@ void write(SessionZooKeeperClient zooKeeperClient, ApplicationId applicationId,
                                                              dockerImageRepository,
                                                              athenzDomain,
                                                              quota,
+                                                             tenantVaults,
```

Comment:
You need to call `zookeeperClient.writeTenantVaults()` before `if (writeSessionData.value())`
as well, this code is only called if a feature flag is set. This code should be cleaned up and explained more I see, I'll look into that

Reply:
Thanks! Missed that one.